ReactiveCocoa/Swift/Signal.swift

```diff
@@ -449,7 +513,30 @@ public func dematerialize<T, E>(signal: Signal<Event<T, E>, NoError>) -> Signal<
 			case .Interrupted:
 				sendInterrupted(observer)
 			}
-		})
+		}
+	}
+}
+
+extension SignalType where T: EventType, E == NoError {
+	public func dematerialize() -> Signal<T.T, T.E> {
+		return Signal { _ in nil }
+		return Signal { (observer: Event<T.T, T.E>.Sink) in
+			return self.signal.observe { event in
+				switch event {
+				case let .Next(innerEvent):
+					observer(innerEvent.event)
+
+				case .Error:
+					fatalError()
```

Comment:
I know this is existing code, but maybe do `fatalError("This can't happen")` to make it clearer?